src/main/java/uk/gov/companieshouse/itemgroupworkflowapi/controller/ItemGroupController.java

```diff
@@ -13,96 +14,118 @@
+     * Item group created, on success return CREATED, on error return BAD_REQUEST<p>
+     */
+    @PostMapping(path = "${uk.gov.companieshouse.itemgroupworkflowapi.createitemgroup}",
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<Object> createItemGroup(final @RequestHeader(X_REQUEST_ID_HEADER_NAME) String xRequestId,
                                                   final @RequestBody ItemGroupData itemGroupData) {
-        logRequestId(requestId);
-        List<String> errors = itemGroupsValidator.validateCreateItemPayload(itemGroupData);
+        logRequestId(xRequestId);
+        List<String> errors = itemGroupsValidator.validateCreateItemData(itemGroupData);
 
         if (!errors.isEmpty()) {
-            return buildValidationResponse(errors, itemGroupData);
+            return buildValidationResponse(xRequestId, errors);
         }
 
         try {
-            if (itemGroupsService.doesItemGroupExist(itemGroupData)) {
-                return buildItemAlreadyExistsResponse(itemGroupData);
-            }
+            if (itemGroupsService.doesItemGroupExist(itemGroupData))
+                return buildItemAlreadyExistsResponse(xRequestId, itemGroupData);
 
             final ItemGroup savedItem = itemGroupsService.createItemGroup(itemGroupData);
-            return buildCreateSuccessResponse(savedItem);
-
-        } catch (Exception ex) {
-            return buildErrorResponse(ex, itemGroupData);
+            return buildCreateSuccessResponse(xRequestId, savedItem);
+        }
+        catch (Exception ex) {
+            return buildErrorResponse(xRequestId, ex, itemGroupData);
         }
     }
 
-    private void logRequestId(String requestId) {
-        Map<String, Object> logMap = logger.createLogMap();
-        logMap.put(REQUEST_ID_LOG_KEY, requestId);
-        logger.getLogger().info("create item group: request id", logMap);
-    }
+    private void logRequestId(String xRequestId) {
+        DataMap dataMap = new DataMap.Builder()
+            .requestId(xRequestId)
+            .build();
 
-    private ResponseEntity<Object> buildCreateSuccessResponse(final ItemGroup savedItem) {
+        logger.getLogger().info(REQUEST_ID_PREFIX, dataMap.getLogMap());
+    }
+    /**
+     * @return HttpStatus.CREATED
+     */
+    private ResponseEntity<Object> buildCreateSuccessResponse(String xRequestId,
+                                                              final ItemGroup savedItem) {
         DataMap dataMap = new DataMap.Builder()
+            .requestId(xRequestId)
             .orderId(savedItem.getData().getOrderNumber())
             .build();
 
-        logger.getLogger().info(CREATE_ITEM_GROUP_CREATED, dataMap.getLogMap());
+        logger.getLogger().info(CREATE_ITEM_GROUP_CREATED_PREFIX, dataMap.getLogMap());
         return ResponseEntity.status(CREATED).body(savedItem);
```

Comment:
As previously discussed, this should probably be populating the response payload with the nested `ItemGroupData` object rather than its nesting `ItemGroup` object, to avoid returning metadata in the response.

Reply:
Looks good to me. 👍